oneview/data_source_logical_enclosure.go

```diff
@@ -0,0 +1,244 @@
+// (C) Copyright 2018 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package oneview
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"strconv"
+)
+
+func dataSourceLogicalEnclosure() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceLogicalEnclosureRead,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"ambient_temperature_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
For Data Sources, all non-required attributes must be `Computed: true,`
```suggestion
				Computed: true,
```

Reply:
Changed all non-required attributes as per the suggestion